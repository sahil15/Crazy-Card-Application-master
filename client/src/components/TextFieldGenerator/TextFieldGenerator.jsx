import React from "react";
import CustomTextField from "../CustomTextField/CustomTextField.jsx";

const TextFieldGenerator = (props) => {
  return (
    <>
      {props.textFields.map((field) => (
        <CustomTextField
          field={field}
          className="center"
          onChange={(e) => onChange(e, field.name)}
          key={field.name}
        ></CustomTextField>
      ))}
    </>
  );
};

export default TextFieldGenerator;
