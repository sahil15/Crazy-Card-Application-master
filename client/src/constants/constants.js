export const DETAILS = {
  textFields: [
    {
      name: "dob",
      label: "Birthday",
      type: "date",
      defaultValue: "2017-05-24",
    },
    {
      name: "address",
      label: "Address",
      minLength: 5,
      type: "text",
      helperText: "Address should be atleast 5 characters long",
    },
    {
      name: "city",
      label: "City",
      type: "text",
      minLength: 3,
      helperText: "City should be atleast 5 characters long",
    },
    {
      name: "postal",
      label: "Postal",
      type: "text",
      helperText: "Postal code should be atleast 5 characters long",
    },
    {
      name: "income",
      label: "Annual Income",
      type: "number",
      required: true,
      helperText: "Must be a number",
    },
  ],
};
export const CARD_DETAILS = [
  {
    name: "apr",
    text: "Representative % APR (variable)",
  },
  {
    text: "Balance Transfer Offer Duration",
    name: "balanceOfferDuration",
  },
  {
    text: "Purchase Offer Duration",
    name: "purchaseOfferDuration",
  },
  {
    text: "Credit Available",
    name: "credit",
  },
];

export const EMP_STATUS_LABEL = "Employment Status";
export const TITLE_LABEL = "Title";
export const TITLE_DROPDOWN = ["Mr", "Mrs", "Ms"];
export const EMP_STATUS_DROPDOWN = ["Full Time", "Part Time", "Student"];
export const HOME_BTN_TEXT = "Navigate to Home";
export const NO_CARD_TEXT = "No Card Selected";
export const INCOME_ERROR_MSG = "Must be a number";
export const INCOME_REQUIRED_MSG = "Required";
export const MIN_LENGTH_MSG = "Length must be atleast 3 characters long! ";

export const FETCH_CARDS_FAILED_MSG = "Fetch Cards Action Failed";
export const FETCH_AVAILABLE_CARDS_FAILED_MSG = "Fetch Available Card Failed";

export const baseURL = "http://localhost:3000";
