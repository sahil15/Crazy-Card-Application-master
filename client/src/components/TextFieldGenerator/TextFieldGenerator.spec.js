import React from "react";
import { shallow } from "enzyme";
import TextFieldGenerator from "./TextFieldGenerator";
import {DETAILS} from "../../constants/constants";

describe("Tests for Card", () => {

  it("Renders correctly", () => {
    const wrapper = shallow(
      <TextFieldGenerator textFields={DETAILS.textFields}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
