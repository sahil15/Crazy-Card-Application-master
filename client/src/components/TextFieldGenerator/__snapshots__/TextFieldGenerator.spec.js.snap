// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Card Renders correctly 1`] = `
<Fragment>
  <Connect(CustomTextField)
    className="center"
    field={
      Object {
        "defaultValue": "2017-05-24",
        "label": "Birthday",
        "name": "dob",
        "type": "date",
      }
    }
    key="dob"
    onChange={[Function]}
  />
  <Connect(CustomTextField)
    className="center"
    field={
      Object {
        "helperText": "Address should be atleast 5 characters long",
        "label": "Address",
        "minLength": 5,
        "name": "address",
        "type": "text",
      }
    }
    key="address"
    onChange={[Function]}
  />
  <Connect(CustomTextField)
    className="center"
    field={
      Object {
        "helperText": "City should be atleast 5 characters long",
        "label": "City",
        "minLength": 3,
        "name": "city",
        "type": "text",
      }
    }
    key="city"
    onChange={[Function]}
  />
  <Connect(CustomTextField)
    className="center"
    field={
      Object {
        "helperText": "Postal code should be atleast 5 characters long",
        "label": "Postal",
        "name": "postal",
        "type": "text",
      }
    }
    key="postal"
    onChange={[Function]}
  />
  <Connect(CustomTextField)
    className="center"
    field={
      Object {
        "helperText": "Must be a number",
        "label": "Annual Income",
        "name": "income",
        "required": true,
        "type": "number",
      }
    }
    key="income"
    onChange={[Function]}
  />
</Fragment>
`;
